packages/react-aria-components/src/ColorThumb.tsx

```diff
@@ -0,0 +1,107 @@
+import {Color} from '@react-types/color';
+import {filterDOMProps} from '@react-aria/utils';
+import {HoverEvents} from '@react-types/shared';
+import {mergeProps} from 'react-aria';
+import React, {createContext, ForwardedRef, forwardRef, HTMLAttributes, InputHTMLAttributes, RefObject, useContext} from 'react';
+import {RenderProps, useRenderProps} from './utils';
+import {useFocusRing} from '@react-aria/focus';
+import {useHover} from '@react-aria/interactions';
+
+interface ColorState {
+  getDisplayColor(): Color,
+  isDragging: boolean
+}
+
+interface InternalColorThumbContextValue {
+  state: ColorState,
+  thumbProps: HTMLAttributes<HTMLElement>,
+  inputXRef: RefObject<HTMLInputElement>,
+  inputYRef?: RefObject<HTMLInputElement>,
+  xInputProps: InputHTMLAttributes<HTMLInputElement>,
+  yInputProps?: InputHTMLAttributes<HTMLInputElement>,
+  isDisabled?: boolean
+}
+
+export const InternalColorThumbContext = createContext<InternalColorThumbContextValue | null>(null);
```

Comment:
Ideally this context would be exported/not internal. However the API for the context is a bit abnormal in that it takes a lot of stuff that isn't just the same as the props, including props/refs for two hidden inputs. If we want to export it we would need to re-think this.